client/src/components/NavBar/index.tsx

```diff
@@ -0,0 +1,285 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { useState, useContext } from 'react';
+import {
+  AppBar, Toolbar, Box, IconButton, Typography,
+  Menu, Container, Avatar, Button, Tooltip, MenuItem,
+} from '@mui/material';
+
+import MenuIcon from '@mui/icons-material/Menu';
+import AdbIcon from '@mui/icons-material/Adb';
+import { useNavigate, NavLink } from 'react-router-dom';
+import { UserContext } from '../../context';
+import { UserContextTypeWithDispatch } from '../../interfaces';
+import httpInstance from '../../services/axiosConfig';
+import CustomizedSnackbars from '../snackbar';
+
+function NavBar() {
+  const [anchorElNav, setAnchorElNav] = useState<null | HTMLElement>(null);
+  const [anchorElUser, setAnchorElUser] = useState<null | HTMLElement>(null);
+  const { userInfo, setUserInfo }:UserContextTypeWithDispatch = useContext(UserContext);
+  const [snackBarProperties, setSnackBarProperties] = useState<
+  { open:boolean, message:string, type:'success' | 'error' }>({ open: false, message: '', type: 'error' });
+
+  const pages = userInfo?.role.toLowerCase() === 'admin'
+    ? [{ title: 'HOME', path: '' }, { title: 'SHOP', path: 'cars' }, { title: 'Dashboard', path: 'admin' }]
+    : [{ title: 'HOME', path: '' }, { title: 'SHOP', path: 'cars' }, { title: 'Contact', path: '/contact' }];
+
+  const settings = userInfo?.role.toLowerCase() === 'admin'
+    ? [{ title: 'Logout', path: 'logout' }]
+    : [{ title: 'Profile', path: 'profile' }, { title: 'Logout', path: 'logout' }];
```

Comment:
as logut will not direct me to `logout`, it will send request to /api/logout , SO add a button in settings list to send a request to `/logout`,
No need to define it here